extensions/amp-analytics/0.1/visibility-model.js

```diff
@@ -440,6 +456,46 @@ export class VisibilityModel {
         (this.maxContinuousVisibleTime_ <= this.spec_['continuousTimeMax']);
   }
 
+  /**
+   * Set the amount that the user had scrolled down the page at the time of
+   * page loading.
+   * @param {number} depth
+   */
+  maybeSetInitialScrollDepth(depth) {
+    if (!this.initialScrollDepthAlreadySet_) {
+      this.initialScrollDepth_ = depth;
+      this.initialScrollDepthAlreadySet_ = true;
+    }
+    this.maybeUpdateMaxScrollDepth(depth);
+  }
+
+  /**
+   * Gets the amount that the user had scrolled down the page, at the time of
+   * ini-load.
+   * @return {number} depth
+   */
+  getInitialScrollDepth() {
+    return this.initialScrollDepth_;
+  }
+
+  /**
+   * Update the maximum amount that the user has scrolled down the page.
+   * @param {number} depth
+   */
+  maybeUpdateMaxScrollDepth(depth) {
```

Comment:
Question: what is our decision on the maxScrollDepth? 
Do we agree to always return the maxScrollDepth or the maximum of maxScrollDepth and elementY.

Reply:
I am waiting for an answer on this - will update you when I receive it.

Reply:
@zhouyx I received the answer today. We will include an enum indicating ABOVE/AT/BELOW the ad, not the integer number of pixels. I will update the PR with this new behaviour.